Sources/Player/Common/Logging/PlayerLoggable.swift

```diff
@@ -0,0 +1,337 @@
+	case migrateLegacyDirectoryFailed(error: Error)
+	case writeEventNotAuthorized
+	case writeEventFailed(error: Error)
+	case sendToEventProducerSerializationFailed
+	case sendToEventProducerFailed(error: Error)
+	case sendEventsNotAuthorized
+	case sendEventsFailed(error: Error)
+	case urlForDirectoryFailed(error: Error)
+	case initializeDirectoryFailed(error: Error)
+
+	// MARK: Streaming Privileges
+	case streamingNotifyNotAuthorized(error: Error)
+	case streamingConnectNotAuthorized(error: Error)
+	case webSocketSendMessageFailed(error: Error)
+	case webSocketReceiveMessageFailed(error: Error)
+	case webSocketHandleErrorSleepAndReconnectionFailed(error: Error) // swiftlint:disable:this identifier_name
+
+	// MARK: Playback Info Fetcher
+	case getPlaybackInfoFailed(error: Error)
+
+	// MARK: FairPlay License Fetcher
+	case getLicenseFailed(error: Error)
+
+	// MARK: Credentials Provider
+	case getAuthBearerTokenCredentialFailed(error: Error)
+	case getAuthBearerTokenToBearerTokenFailed
+
+	// MARK: Credentials Success Data Parser
+	case credentialsSuccessParserParsingFailed(error: Error)
+
+	// MARK: Download Task
+	case downloadFailed(error: Error)
+	case downloadFinalizeFailed(error: Error)
+
+	// MARK: Offline Engine
+	case deleteOfflinedItem(error: Error)
+
+	// MARK: Asset Cache Legacy
+	case deleteAssetCacheFailed(error: Error)
+
+	// MARK: StoredLicenseLoader
+	case licenseLoaderContentKeyRequestFailed(error: Error)
+	case licenseLoaderProcessContentKeyResponseFailed(error: Error) // swiftlint:disable:this identifier_name
+
+	// MARK: StreamingLicenseLoader
+	case streamingGetLicenseFailed(error: Error)
+
+	// MARK: HttpClient
+	case payloadEncodingFailed(error: Error)
+	case payloadDecodingFailed(error: Error)
+
+	// MARK: URLSession
+	case loadDataForRequestFailed(error: Error)
+
+	// MARK: ResponseHandler
+	case backoffHandleResponseFailed(error: Error)
+
+	// MARK: Downloader
+	case startDownloadFailed(error: Error)
+
+	// MARK: LicenseDownloader
+	case licenseDownloaderContentKeyRequestFailed(error: Error)
+	case licenseDownloaderGetLicenseFailed(error: Error)
+
+	// MARK: MediaDownloader
+	case downloadFinishedMovingFileFailed(error: Error)
+
+	// MARK: AVQueuePlayerWrapperLegacy
+	case legacyReadPlaybackMetadataFailed(error: Error)
+
+	// MARK: AVQueuePlayerWrapper
+	case readPlaybackMetadataFailed(error: Error)
+
+	// MARK: DJProducer
+	case djSessionStartFailed(error: Error)
+	case djSessionSendCommandFailed(error: Error)
+
+	// MARK: InternalPlayerLoader
+	case loadUCFailed(error: Error)
+
+	// MARK: PlayerEngine
+	case loadPlayerItemFailed(error: Error)
+}
+
+// MARK: - Logging
+extension PlayerLoggable {
+
+	var loggingMessage: Logger.Message {
+		return switch self {
+		// Event Sender
+		case .sendEventOfflinePlayFailed:
+			"EventSender-sendEventOfflinePlayFailed"
+		case .sendLegacyEventFailed:
+			"EventSender-sendLegacyEventFailed"
+		case .migrateLegacyDirectoryFailed:
+			"EventSender-migrateLegacyDirectoryFailed"
```

Comment:
Can some of the name be generated automatically from the class name or something like that?

Reply:
I think we can get from the class name, I'll check that. As for the func, I'm not so sure. If we use #func, it returns the whole naming in the func and a specific format. Maybe with some formatting we could do it, to get it until the first parenthesis. 🤓 Do you mind if I add a ticket for this to be checked separately?